synapse/config/_base.pyi

```diff
@@ -17,6 +16,7 @@ from typing import (
 )
 
 import jinja2
+from typing_extensions import Literal
```

Comment:
This PR is no longer necessary because we're now targeting Python 3.8 as the minimum version which is where `typing.Literal` was introduced.

But perhaps we want to do this solely to align our single `typing.Literal` usage to `typing_extensions.Literal` which we do everywhere else.

Or maybe we want to instead move everything to `typing.Literal` :fast_forward: 

Pokes in whatever direction welcome

Reply:
I think move to `typing.Literal` ASAP when we drop 3.7 is the way to go